politeiad/backendv2/tstorebe/tstorebe.go

```diff
@@ -0,0 +1,1251 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package tstorebe
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/decred/dcrd/chaincfg/v3"
+	"github.com/decred/politeia/politeiad/api/v1/mime"
+	backend "github.com/decred/politeia/politeiad/backendv2"
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/plugins"
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/tstore"
+	"github.com/decred/politeia/util"
+	"github.com/subosito/gozaru"
+)
+
+var (
+	_ backend.Backend = (*tstoreBackend)(nil)
+)
+
+// tstoreBackend implements the Backend interface using a tstore as the
+// data store.
+type tstoreBackend struct {
+	sync.RWMutex
+	appDir   string
+	dataDir  string
+	shutdown bool
+	tstore   *tstore.Tstore
+
+	// tokens contains the short token to full token mappings. The
+	// short token is the first n characters of the hex encoded record
+	// token, where n is defined by the short token length politeiad
+	// setting. Record lookups using short tokens are allowed. This
+	// cache is used to prevent collisions when creating new tokens
+	// and to facilitate lookups using only the short token. This cache
+	// is built on startup.
+	tokens map[string][]byte // [shortToken]fullToken
```

Comment:
Are we moving this cache out of memory and make it generic?

Reply:
Yes, the plan is to nuke all filesystem and memory caches.